src/debugging/inspector/Inspector/NativeScript Inspector/Communication.h

```diff
@@ -0,0 +1,41 @@
+#ifndef Communication_h
+#define Communication_h
+
+#import <Foundation/Foundation.h>
+#import <netinet/in.h>
+#import <sys/types.h>
+#import <sys/socket.h>
+#import <errno.h>
+#import <stdlib.h>
+#import <string.h>
+#import <notify.h>
+
+#define CheckError(retval, handler)                             \
+    ({                                                          \
+        typeof(retval) errorCode = retval;                      \
+        BOOL success = NO;                                      \
+        if (errorCode == 0)                                     \
+            success = YES;                                      \
+        else if (errorCode == -1)                               \
+            errorCode = errno;                                  \
+        if (!success)                                           \
+            handler([NSError errorWithDomain:NSPOSIXErrorDomain \
+                                        code:errorCode          \
+                                    userInfo:nil]);             \
+        success;                                                \
+    })
+
+typedef void (^InspectorReadHandler)(dispatch_data_t data);
+typedef void (^InspectorErrorHandler)(NSError* error);
+
+@interface TNSCommunicationChannel : NSObject
+@property(nonatomic) dispatch_fd_t socket;
+@property(nonatomic, strong) dispatch_io_t ioChannel;
+
+- (id)initWithSocketPath:(NSString*)socketPath readHandler:(InspectorReadHandler)readHandler errorHandler:(InspectorErrorHandler)errorHandler;
+- (void)sendMessage:(uint32_t)length message:(void*)message;
+- (void)dealloc;
```

Comment:
You don't need to add the `dealloc` method in `@interface` in order to override it.